vm/src/iter.rs

```diff
@@ -33,7 +33,11 @@ pub trait PyExactSizeIterator<'a>: ExactSizeIterator<Item = &'a PyObjectRef> + S
         let rhs = other;
         let lhs_len = lhs.len();
         let rhs_len = rhs.len();
+
         for (a, b) in lhs.zip(rhs) {
+            if a.rich_compare_bool(b, PyComparisonOp::Eq, vm)? {
```

Comment:
I'd go with `vm.bool_eq(a, b)` here, practically the same

Reply:
I fixed the code.